src/twisted/names/test/test_names.py

```diff
@@ -1221,13 +1223,50 @@ def loadBindString(self, s):
         @return: a new bind authority
         @rtype: L{twisted.names.authority.BindAuthority}
         """
-        fp = FilePath(self.mktemp().encode("ascii"))
+        fp: FilePath[AnyStr] = FilePath(path)
         fp.setContent(s)
 
         return authority.BindAuthority(fp.path)
```

Comment:
The problem here is that `FileAuthority` doesn't document what the `filename` parameter to its `__init__` is. Either nailing it down to `str` or making `FileAuthority` *itself* `Generic[AnyStr]` would allow this to be addressed more comprehensively.

That isn't required to land this since it might end up touching a bunch more files, but I think we should file it as a follow-up if not.

Reply:
Once I land this, I will create a new issue to address this.